src/Extension/CoreExtension.php

```diff
@@ -739,9 +741,13 @@ function twig_last(Environment $env, $item)
  */
 function twig_join_filter($value, $glue = '', $and = null)
 {
+    if (!twig_test_iterable($value)) {
+        $value = (array) $value;
```

Comment:
I don't like that, but this was in the original code before the introduction of `twig_to_array()`, so I kept it here.